### PR TITLE
test: shallowClone in checkstyle-tester with SHA checkout

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Issue #818: Aims to see if the shallowClone feature works fine

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
@nrmancuso, i was trying to test my changes [Checkstyle/](https://github.com/checkstyle/contribution/pull/843/commits/90c0a5789cfb968cbf895ead00d612fb97b6878a) for shallow-cloning. But, I assume we need to create a separate testing branch for the same. Is there any other way?

**Comment by nrmancuso** [comment]
@relentless-pursuit there are a bunch of ways to test this out, but I would just use your fork of contribution in the `checkout_from` command

**Comment by relentless-pursuit** [comment]
> @relentless-pursuit there are a bunch of ways to test this out, but I would just use your fork of contribution in the `checkout_from` command

Thank you. It seems to be working. Currently, analysing the reports. And, I can see 
ci/circleci: no-exception-Pmd-elasticsearch-lombok-ast report succeeding, while retaining specific checkout option. And, I assume the code was more or less the same, resue of Romani's PR. So, hopefully, I can have some idea if the code changes are working.

**Comment by nrmancuso** [comment]
@relentless-pursuit I am placing blocked label and putting this into draft mode to make it obvious that we don't need to remove this.

**Comment by romani** [comment]
Rebased to show latest execution results.
@rnveach , fyi

**Comment by romani** [comment]
Rebase to show latest execution results.
@rnveach , fyi

**Comment by relentless-pursuit** [comment]
@rnveach based on the latest rebasing, i would highlight for your reference:
Successfull shallow clone: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/25480/workflows/b006820e-4d74-4fba-817f-a5693176a112/jobs/588430
```
Testing Checkstyle started
   [delete] Deleting directory /home/circleci/project/.ci-temp/contribution/checkstyle-tester/src/main/java
Shallow clone git repository 'apache-struts' to repositories/apache-struts folder ...
Cloning into 'repositories/apache-struts'...
Cloning git repository 'apache-struts' - completed
```

Successful build while retatining checkout to specific SHA: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/25480/workflows/b006820e-4d74-4fba-817f-a5693176a112/jobs/588425
```
Testing Checkstyle started
   [delete] Deleting directory /home/circleci/project/.ci-temp/contribution/checkstyle-tester/src/main/java
Cloning git repository 'pmd' to repositories/pmd folder ...
Cloning into 'repositories/pmd'...
Cloning git repository 'pmd' - completed

Running command: git fetch --tags
Resetting git sources to commit 'pmd_releases/6.21.0'
Running command: git reset --hard refs/tags/pmd_releases/6.21.0
HEAD is now at a28e9e22e5 [maven-release-plugin] prepare release pmd_releases/6.21.0
pmd is synchronized
     [copy] Copying 4019 files to /home/circleci/project/.ci-temp/contribution/checkstyle-tester/src/main/java/pmd
Running 'mvn clean' on src/main/java ...
Running command: mvn -e --no-transfer-progress --batch-mode clean
```

**Comment by romani** [comment]
Please fix CI problems https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=21366&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1325
It is better to have fully green CI, as good sign. Red CI is red flag for all maintainers and suspicious filling is growing

**Comment by romani** [comment]
@relentless-pursuit, contribution PR is closed, please prepare this PR for merge.

**Comment by relentless-pursuit** [comment]
> some minor revert for repo name will be required

@romani can you help me understand what needs to be changed in this repo? Also, i used the parameter `useShallowClone` in `validation.sh`.

File: .ci/validation.sh [hunk 1]
```diff
@@ -895,6 +895,7 @@ no-exception-struts)
   sed -i'' 's/#apache-struts/apache-struts/' projects-to-test-on.properties
   groovy ./diff.groovy --listOfProjects projects-to-test-on.properties \
       --patchConfig checks-nonjavadoc-error.xml  -p "$BRANCH" -r ../../..  \
+      --useShallowClone \
```
Comment (romani) [inline .ci/validation.sh:898]: There are 13 usages of diff.groovy please update all of them

**Review by romani — CHANGES_REQUESTED** [review]
>you help me understand what needs to be changed in this repo?

See below.

Items

File: .ci/validation.sh (hunk 1 above)
Comment (relentless-pursuit) [reply .ci/validation.sh:898]: Done

**Comment by relentless-pursuit** [comment]
> @relentless-pursuit, contribution PR is closed, please prepare this PR for merge.
> 
> Commit prefix should be "Pull #14662: ...."

Done

**Review by romani — APPROVED** [review]
Ok to merge.

Thanks a lot for finishing long update

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**